en/lessons/specifics/benchmarking.md

```diff
@@ -0,0 +1,346 @@
+    * **best** - the most frequent best fit unit will be used. A tie will result in the larger unit being selected.
+    * **largest** - the largest best fit unit will be used
+    * **smallest** - the smallest best fit unit will be used
+    * **none** - no unit scaling will occur. Durations will be displayed in microseconds, and ips counts will be displayed without units.
+
+## Inputs
+
+It's very important to benchmark your functions with data that reflects what that function might actually operate on in the real world. Frequently a function can have very different behavior on small sets of data versus large sets of data! This is where Benchee's `inputs` configuration option comes in. This allows you to test the same function but with as many different functions as you like, and then you can see the results of the benchmark with each of those functions.
+
+So, let's look at our original example again:
+
+```elixir
+list = Enum.to_list(1..10_000)
+map_fun = fn(i) -> [i, i * i] end
+
+Benchee.run(%{
+  "flat_map"    => fn -> Enum.flat_map(list, map_fun) end,
+  "map.flatten" => fn -> list |> Enum.map(map_fun) |> List.flatten end
+})
+```
+
+In that example we're only using a single list of the integers from 1 to 10,000. Let's update that to use a couple different inputs so we can see what happens with smaller and larger lists. So, open that file, and we're going to change it to look like this:
+
+```elixir
+map_fun = fn(i) -> [i, i * i] end
+inputs = %{
+  "small list" => Enum.to_list(1..100),
+  "medium list" => Enum.to_list(1..10_000),
+  "large list" => Enum.to_list(1..1_000_000)
+}
+
+Benchee.run(%{
+  "flat_map"    => fn(list) -> Enum.flat_map(list, map_fun) end,
+  "map.flatten" => fn(list) -> list |> Enum.map(map_fun) |> List.flatten end
+}, inputs: inputs)
+```
+
+You'll notice two differences. First, we now have an `inputs` map that contains the information for our inputs to our functions. We're passing that inputs map as a configuration option to `Benchee.run/2`.
+
+And since our functions need to take an argument now, we need to update our benchmark functions to accept an argument, so instead of:
+```elixir
+fn -> Enum.flat_map(list, map_fun) end
+```
+
+we now have:
+```elixir
+fn(list) -> Enum.flat_map(list, map_fun) end
+```
+
+Let's run this again using:
+
+```shell
+$ mix run benchmark.exs
+```
+
+Now you should see output in your console like this:
+
+```shell
+Operating System: macOS
+CPU Information: Intel(R) Core(TM) i5-4260U CPU @ 1.40GHz
+Number of Available Cores: 4
+Available memory: 8.589934592 GB
+Elixir 1.5.1
```

Comment:
Probably we could use the [version placeholders](https://github.com/elixirschool/elixirschool/blob/master/_config.yml#L43-L48) within [lines 194-195](https://github.com/elixirschool/elixirschool/pull/1151/files#diff-5364d6d75e79d566801405a1097e95b8R194) and [332-333](https://github.com/elixirschool/elixirschool/pull/1151/files#diff-5364d6d75e79d566801405a1097e95b8R332).

Reply:
Sorry, forgot to explain my reasoning. Since these benchmarks are snapshots of output, I didn't think it would make sense for them to have those variable version numbers. It's reasonable to assume that in 2 years this output might change, and if we're showing Elixir 2.0.1 with this output, I would think it might be confusing.

That said, if you feel strongly about it I'm happy to use the placeholders. It just made sense to me to not use them there, but I don't feel that strongly about it.